src/modify_resources.py

```diff
@@ -82,232 +86,239 @@ def remove_all_network_manipulation(ssh_client, remove_all_machines=False):
-	if num_stress < 0:
-		invalid_resource_parameter("CPU Allocation", limit)
-		return
+    """CPU Limit describes how much CPU will be used by the stresser"""
+    if num_stress < 0:
+        invalid_resource_parameter("CPU Allocation", limit)
+        return
         
         #Separate CPU limit into several smaller increments of 10 instead of one larger one
-        for x in range(num_stress):
-	        cmd = "stress -c 1 &> /dev/null & cpulimit -p $( pidof -o $! stress ) -l {} &> /dev/null &".format(10)
-	        ssh_exec(ssh_client, cmd)
+    for x in range(num_stress):
+        cmd = "stress -c 1 &> /dev/null & cpulimit -p $( pidof -o $! stress ) -l {} &> /dev/null &".format(10)
+        ssh_exec(ssh_client, cmd)
 
 #Allows you to set the CPU periods in a container, intended for a container that is already running
 #as opposed to update_cpu which actually just stresses the CPU by a predetermined amount
 #Assumes that that CPU_period and CPU_quota were not set beforehand
 #CPU period is assumed to be 1 second no matter what
 def set_cpu_shares(ssh_client, cpu_quota):
-        container_names = get_container_names(ssh_client)
-        cpu_quota = int(cpu_quota * get_num_cores(ssh_client) * 1000000)
-        cpu_period = 1000000 #1 second in microseconds
-
-        #CPU Quota must be less than 1 second
-        while cpu_quota > 1000000:
-                cpu_quota = cpu_quota/2
-                cpu_period = cpu_period/2
-
-        print 'Adjusted CPU period: {}'.format(cpu_period)
-        print 'CPU Quota: {}'.format(cpu_quota)
-
-        throttled_containers = []
-        
-        for container_name in container_names:
-                if container_name not in container_blacklist:
-                        update_command = 'docker update --cpu-period {} --cpu-quota {} {}'.format(cpu_period, cpu_quota, container_name)
-                        print update_command
-                        ssh_client.exec_command(update_command)
-                        throttled_containers.append(container_name)
+    container_names = get_container_names(ssh_client)
+    # cpu_quota = int(cpu_quota * get_num_cores(ssh_client) * 1000000)
+    cpu_quota = int((float(cpu_quota) / 10) * 1000000)
+    cpu_period = 1000000 #1 second in microseconds
+
+    #CPU Quota must be less than 1 second
+    # while cpu_quota > 1000000:
+    #         cpu_quota = cpu_quota/2
+    #         cpu_period = cpu_period/2
+
+    print 'Adjusted CPU period: {}'.format(cpu_period)
+    print 'CPU Quota: {}'.format(cpu_quota)
+
+    throttled_containers = []
+    
+    core = 0
+    for container_name in container_names:
+        if container_name not in container_blacklist:
+            update_command = 'docker update --cpu-period {} --cpu-quota {} --cpuset-cpus {} {}'.format(cpu_period, cpu_quota, core, container_name)
+            print update_command
+            ssh_client.exec_command(update_command)
+            throttled_containers.append(container_name)
+            core += 1
```

Comment:
Is the assumption that each container is only pinned to a single core? Also, you should not assume that container_name[0] is assigned to the 0th core, container_name[1] is assigned to the 1st core, etc. Why update the cpuset anyways?